main.cpp

```diff
@@ -56,46 +56,107 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    float value;
+    const char* name;
+    T(const float v, const char* ptr) :  //1
+    value(0.0f),//2
+    name(nullptr)//3
+    {
+        value = v;
+        if(ptr != nullptr)
+        {
+            name = ptr;
+        }
+        else std::cout << "name is nullptr!\n";
+    }
 };
 
-struct <#structName1#>                                //4
-{
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+struct Comparer                                //4
+{   
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a != nullptr && b != nullptr)
+        {
+            if( a->value < b->value ) return a;
+            if( a->value > b->value ) return b;
+            std::cout << "a == b!\n";
+            return nullptr;
+        }
+        std::cout << "a and/or b are nullptr!\n";
         return nullptr;
+        
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    float x { 0 }, y { 0 };
+    float bringTogether(float* updatedValue)      //12
     {
-        
+        if( updatedValue != nullptr)
+        {
+            std::cout << "U's x value: " << this->x << std::endl;
+            this->x = (*updatedValue);
+            std::cout << "U's x updated value: " << this->x << std::endl;
+            std::cout << "bringing x and y together\n";
+            while( std::abs(this->y - this->x) > 0.001f )
+            {
+                if (this->y < this->x)
+                {
+                    this->y += .1f;
+                    this->x -= .1f;
+                }
+                else if(this->y > this->x)
+                {
+                    this->y -= .1f;
+                    this->x += .1f;
+                }
+                else std::cout << "x == y!\n";
```

Comment:
If you use `{ }` in any `if/else` blocks, use them for EVERY block in that `if/else if/else` logic.
- [ ] revise